From 31ca19eb99dbd588a1995898ab1baf6cef2180d6 Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Tue, 29 Nov 2016 16:59:48 +0100
Subject: [PATCH] pull: Exit early on error without aborting transaction

At this point we have not set up repo yet, so aborting would crash.
---
 common/flatpak-dir.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
index 8213440..1094cf4 100644
--- a/common/flatpak-dir.c
+++ b/common/flatpak-dir.c
@@ -1563,13 +1563,13 @@ flatpak_dir_pull (FlatpakDir          *self,
   g_autoptr(GPtrArray) subdirs_arg = NULL;
 
   if (!flatpak_dir_ensure_repo (self, cancellable, error))
-    goto out;
+    return FALSE;
 
   if (!ostree_repo_remote_get_url (self->repo,
                                    repository,
                                    &url,
                                    error))
-    goto out;
+    return FALSE;
 
   if (*url == 0)
     return TRUE; /* Empty url, silently disables updates */
